apps/dashboard/src/data/projects/actions.js

```diff
@@ -43,7 +43,7 @@ export function* saveAndUpdateProject( projectId, project ) {
 			? yield dispatchAsync( patchProject, [ projectId, project ] )
 			: yield dispatchAsync( createProject, [ project ] );
 
-		yield redirect( `/edit/${ response.data.id }` );
+		yield redirect( `/project/${ response.data.id }` );
```

Comment:
I believe calling that after `updateProject` not only makes more sense but also fixes the issue where the editor temporarily flickers the loading placeholder.

Also, not sure if `yield` is actually necessary anymore since this doesn't return an action anymore.

Reply:
Maybe I'm not getting it right but, the redirect is called AFTER the `updateProject`, did you mean BEFORE?

On the other hand, the `updateProject` is actually to blame for the full re-render, but that's mostly on behalf of a flawed flow/sequence (the one I'm trying to address and, I think, out of this scope)

Reply:
Right now, it's:

1. Making the request
2. Triggering the redirect
3. Updating the state with the new data

2. and 3. need to be swapped around.

Reply:
Done. I don't think the flickering will be gone, the problem persists and I think is elsewhere. We'll deal with it

Reply:
Not quite... what I meant was It should be like this:

```javascript
yield updateProject( ... );

redirect( ... );
return;
```

Or even just `return redirect( ... );` - it doesn't really change anything in the handling and is a little bit nicer to read. (Returning undefined means there's nothing to do.)

Reply:
Sorry! Messed up that stash 😬